features/FxA-89-devices-view/README.md

```diff
@@ -0,0 +1,410 @@
+
+When the user is viewing the device list
+from a web session on accounts.firefox.com,
+and is on a compatible Firefox browser
+that is not currently connected to Sync,
+we provide a "Connect..." button
+that will allow them to sign in to sync
+on that device.
+
+<img src="devices-view-current-web-session.png" alt="Zero or one Device Views" />
+
+### Disconnecting View
+
+When the user selects to disconnect a device,
+we overlay a modal confirmation dialog.
+This will minimize accidental clicks on "Disconnect..."
+and help us measure signals as to why a device is removed.
+
+In this view, we should not see
+any message to download Firefox on other devices.
+We want to minimize noise and distractions.
+
+When the user clicks and releases the "Disconnect..." button
+the following occurs:
+
+1. We verify that the user session is recent.
+   If it is too old then we bounce the user
+   through the existing "force auth" screen
+   in order to refresh their login session.
+2. We display a modal dialog
+   asking them to confirm
+   and supply a reason why they are disconnecting:
+
+   <img src="modal-disconnect-dialog.png" />
+
+3. We display an error "Valid response required"
+   if they submit the dialog
+   without selecting a reason:
+
+   <img src="modal-disconnect-dialog-validation-error.png" />
+
+4. When the user selects any option,
+   the placeholder floats up to be a field label
+   and the Disconnect button becomes enabled:
+
+   <img src="modal-disconnect-dialog-filled.png" />
+
+5. When the form is submitted
+   we disconnect the device
+   by calling the `/account/device/destroy` endpoint
+   on the server..
+
+6. Depending on the user's selected reason for disconnecting
+   we may show an additional message styled as follows:
+
+   <img src="modal-disconnect-dialog-success.png" />
+
+The available choices for disconnection reason will be:
+
+* Suspicious device
+  * Upon Disconnect, show additional message with the body copy:
+    “We’re sorry to hear about this. If this was a device you
+    really don’t trust, you should change your Firefox Account password,
+    and change any passwords saved in Firefox.”
+* Lost or stolen device
+  * Upon Disconnect, show additional message with the body copy:
+  “We’re sorry to hear about this. You should change your Firefox Account password,
+  and look for information from your device manufacturer about erasing your data remotely.”
+* Old or replaced device
+  * Upon Disconnect, don’t present an additional message.
+    Close the dialog and return to the device view.
+* Rather not say
+  * Upon Disconnect, don’t present an additional message.
+    Close the dialog and return to the device view.
+
+
+After device removal has been confirmed,
+the dialog closes and,
+the row is overtaken by the area beneath it:
+
+<img src="devices-view-disconnecting.gif" />
+
+One the server side,
+we send the user an email notification about the removed device.
```

Comment:
It's not clear to me why we need this at all. I feel like there is a sentence missing about the goals of this functionality.